### ISSUE TITLE
Chrome not displaying remoteVideoTrack from Android after publish/unpublish/publish it

### ISSUE THREAD

**Issue by DmitryMorozov228** [twilio/twilio-video.js#1096]
- [x] I have verified that the issue occurs with the latest twilio-video.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-video.js/blob/master/CHANGELOG.md).
 - [x] I reviewed the [Common Issues](https://github.com/twilio/twilio-video.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [x] I verified that the [Quickstart](https://github.com/twilio/video-quickstart-js) application works in my environment.
 - [x] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

Steps to reproduce:
1. Publish video track from Android client.
2. Subscribed on the track in the browser. I have received video track with 'isStarted' property = 'false' and attach it to the DOM. After that track.on('started', ()) event was triggered and video was successfully played on the UI.
3. After that Android client unpublished the video track.
4. Unsubscribed from the track on the browser and detach it from the DOM. Video track was detached and removed from the UI.
5. Publish video track from Android client.
6. Subscribed on the track in the browser. I have received video track with 'isStarted' property = 'false' and attach it to the DOM. After that track.on('started', ()) event was not triggered and video wasn't successfully played on the UI. Also here I was got track with dimensions: { width: null, height: null }

**Note:** This problem is not actual for Firefox browser.

Any help would be greatly appreciated, and if there is more information I can provide please let me know.

**Code to reproduce the issue:**

```js
public onStart (container, { activeCall }) {
    return new Promise(resolve => {
      this.container = container
      let remoteAddedTracksCount = 0
      const onTrackSubscribed = track => {
        this.onTrackSubscribed(track)
        this.attachTrack(container, track)
        remoteAddedTracksCount += 1
        if (remoteAddedTracksCount >= REMOTE_TRACKS_COUNT) {
          resolve()
        }
      }
      this.nativeRoom.participants.forEach(participant => {
        this.attachParticipantTracks(participant, container)
        // if refresh page
        participant.on('trackSubscribed', (track: Track, publication: TrackPublication) => {
          onTrackSubscribed(track)
        })
        // if user moved to another page
        const remoteTrackPublications = Array.from<TrackPublication>(participant.tracks.values())
          .filter(publication => !!publication.track)
        if (remoteTrackPublications.length >= REMOTE_TRACKS_COUNT) {
          resolve()
        }
      })
      this.nativeRoom.on('participantConnected', participant => {
        participant.tracks.forEach(publication => {
          // If the TrackPublication is already subscribed to, then attach the Track to the DOM.
          if (publication.track) {
            onTrackSubscribed(publication.track)
          }
          // Once the TrackPublication is subscribed to, attach the Track to the DOM.
          publication.on('subscribed', track => {
            onTrackSubscribed(track)
          })
        })
      })
      this.nativeRoom.localParticipant.audioTracks.forEach(audioTrackPublication => {
        if (_get(activeCall, ['properties', 'isMutedMicrophone'])) {
          audioTrackPublication.track.disable()
        } else {
          audioTrackPublication.track.enable()
        }
      })
    })
  }

private onTrackSubscribed = track => {
    if (track.kind === TRACK_TYPES.data) {
      track.on('message', event => {
        const data = JSON.parse(event)
        const { id } = data

        if (deliveryManager.hasSending(id)) {
          deliveryManager.get(id).resolve()
        } else {
          this.emit('message', data)
          this.onSendToDataTrack({ id: data.id, type: MESSAGE_TYPES.received })
        }
      })
      return
    }
  }

  private attachTrack = (container, track) => {
    if (track._isEnabled) {
      container.appendChild(track.attach())
    }
  }

```

**Expected behavior:**

The 'started' event should be triggered. Remote Video Track should have 'isStarted' property is equal 'true'. The Remove Video Track will  be played on the UI.

**Actual behavior:**

The Remote Video Track has property isStarted is equal 'false'. Event 'started' was not triggered for the track. Cannot see remote video on Chrome after attach it to the DOM. 

**Software versions:**

 - [ Chrome  83.0.4103.116] Browser(s):
 - [ Windows - Android Galaxy Tab S2] Operating System:
 - [ 2.5.1] twilio-video.js [3.2.1] Android Video SDK:
 - [ React] Third-party libraries (e.g., Angular, React, etc.):

**Comment by makarandp0** [comment]
Hello @DmitryMorozov228, Thank you for writing about this issue. 

Can you please post room-sid and track-sid for the room/ track that did not work ? Any logs from client side would be helpful as well.

Thanks,
Makarand

**Comment by DmitryMorozov228** [comment]
Hello @makarandp0, sure!

Room-Sid: RMc64878711d8892a9c57b0d742542adc8
Track-Sid: MTc46c47614446cc0c2833f5fbfa820125

No one error/warning logs were found on client side.

**Comment by makarandp0** [comment]
Hey @DmitryMorozov228, 

I looked at logs for the room and looks like there were lot of participants (36) with one participant on android sdk 3.2.1 and rest on jsdk 2.1.0 ( FYI: it's not a problem, but it wasn't 2.5.1 as you noted in the post - is that expected?)

This is a peer-to-peer room, so we do not have much insights into media from the tracks, but I do see that the track you mentioned was "unpublished" by android participant. I also see that android participant published another video track ( every time track gets published it gets a new track-sid, please let us know track sid that had the issues, and the participant sid which had issue with the track starting. Or did other participants have same issue with the track as well? 

Does this issue repro consistently - if so it would be great if we can see a controlled repro - (with just 2 participants).

Thanks,
Makarand

**Comment by DmitryMorozov228** [comment]
I'm so sorry, you are right. For testing purpose I have reverted twilio-video js version to 2.1.0. But this issue actual for both versions (2.1.0, 2.5.1)
Do you mean concurrent participants? (36) According with twilio specification P2P rooms up to 10 participants.
I'm not sure about count of participants. I can't imagine how it possible. I have one local participant and one remote. Probably it was the same same participant, he just did reconnecting to the room.
I will try to create a new room and then come back to you with details.

Thanks you for quick updates.

**Comment by DmitryMorozov228** [comment]
Hey @makarandp0, you can find details below:
I've created a new room with SID = 'RM536eb202cbf5ff9687fe0855b620eb90'
Participant SID who is publishing/unpublishing video tracks = 'PAa3660c2d742d7a68bf5fc9a54d6b1764'
Participant SID who is trying to attach video track of remote participant = 'PA51960323902b80de373e24aa6024ac40'
Video Track SID that was successfully started in the DOM = 'MT92e16a995dad38411119fc3fadb60c78'
Video Track SID that wasn't successfully started in the DOM = 'MT1a48c106a1c315193c95c5f33621ddd0'

Also I have recorder a GIF with local debugging, probably it would be help you:
[https://drive.google.com/file/d/10lc4RaoMWrT5wEbWdRkQKJjwME_R_rgC/view?usp=sharing](url)

**Comment by DmitryMorozov228** [comment]
Any updates on this?

**Comment by makarandp0** [comment]
@DmitryMorozov228, I took a look at the room logs, and I do see that chrome participant received healthy bytes on the  `MT1a48c106a1c315193c95c5f33621ddd0` track.

Looking at your code I see one  issue: You are not detaching track from the dom element when it gets unsubscribed - so would it be possible previous track is still in the element, and that is causing issues.

That still does not explain why the track did not emit started event - I do not see the code you referenced listening to started event, can you include that code, and make a fix for removing track when it gets unsubscribed/unpublished ?

Let me know how it goes, 
Thanks
Makarand

**Comment by DmitryMorozov228** [comment]
@makarandp0 Thanks for updates! 
A part of code for detaching track from the DOM is here:


```js
    // When a Participant removes a Track, detach it from the DOM.
    room.on('trackUnsubscribed', (track: Track, publication: TrackPublication, participant: Participant) => {
      this.detachTracks([track])
      this.emit('trackUnsubscribed', track, publication, participant)
    })

  public detachTracks(tracks: Track[]) {
    tracks.forEach(track => {
      if (track.kind === 'audio' || track.kind === 'video') {
        track.detach().forEach(detachedElement => {
          detachedElement.remove()
        })
      }
    })
  }
```
Part of code related with listening a started event:

```js
private attachTrack = (container, track) => {
    if (track._isEnabled) {
      if (track.kind === 'video') {
        track.on('started', (track) => {
          console.log('Track is started.')
        })
      }
      container.appendChild(track.attach())
    }
  }
```
Also I have investigated Twilio logs and got next results: 
In the previous version twilio-video-js SDK (1.15.0) video and audio detached tracks were Ended.
In new version twilio-video-js SDK (2.1.0) only audio detached track was Ended. 
I've attached logs below, probably it helps.
[twilio_1.15.0_logs.txt](https://drive.google.com/file/d/1MbtskURLAkmzRbihzM_vfGeEJO-s9guC/view?usp=sharing)
[twilio_2.5.1_logs.txt](https://drive.google.com/file/d/1qUNKQ7zmXAE3_2b2MfKGslUURIEUUTFW/view?usp=sharing)

**Comment by manras16** [comment]
Facing same issue 
When the participant shares the screen and then unshares, the video doesn't gets the dimensions (i.e width and height is null) and isStarted is returning false , isSubscribed is true. It happens only when the participant shares and unshares screen through chrome browser. 
It works when the participant does the same thing through firefox.

twilio video version: "^1.18.2"
Chrome version : Version 84.0.4147.89

**Comment by DmitryMorozov228** [comment]
@makarandp0 Any updates on this?

**Comment by PavloShchur** [comment]
Are there any updates?

**Comment by makarandp0** [comment]
Hey @DmitryMorozov228 sorry for late response - Can you share some room sids and client side logs with the repro again. if possible can you share link to your app so that I can try the repro myself ?

**Comment by DmitryMorozov228** [comment]
Hi @makarandp0,
Room SID = 'RM536eb202cbf5ff9687fe0855b620eb90'
Participant SID who is publishing/unpublishing video tracks = 'PAa3660c2d742d7a68bf5fc9a54d6b1764'
Participant SID who is trying to attach video track of remote participant = 'PA51960323902b80de373e24aa6024ac40'
Video Track SID that was successfully started in the DOM = 'MT92e16a995dad38411119fc3fadb60c78'
Video Track SID that wasn't successfully started in the DOM = 'MT1a48c106a1c315193c95c5f33621ddd0'
Client logs:
[twilio_1.15.0_logs.txt](https://drive.google.com/file/d/1MbtskURLAkmzRbihzM_vfGeEJO-s9guC/view?usp=sharing)

**Comment by makarandp0** [comment]
Hello @DmitryMorozov228, Unfortunately I did not find any records of the room-sid  - We keep records only for last 30 days. Do you have any sids that are more recent?

Also I noticed from the logs that you are using 1.x version of the sdk,Support for 1.x will cease on December 4th, 2020. This branch will only receive fixes for critical issues until that date.  Please upgrade to 2.x. Check [this guide](https://www.twilio.com/docs/video/migrating-1x-2x) when planning your migration to 2.x.

**Comment by adityajoshee** [comment]
I'm facing the same issue with chrome, though it works in firefox. Please help on this @manjeshbhargav
